packages/realm-react/README.md

```diff
@@ -46,69 +44,170 @@ class Task extends Realm.Object {
+            <Pressable
+              onPress={() =>
+                realm.write(() => {
+                  item.isComplete = !item.isComplete
+                })
+              }><Text>{item.isComplete ? "✅" : "☑️"}</Text></Pressable>
+            <Text style={{ paddingHorizontal: 10 }} >{item.description}</Text>
+            <Pressable
+              onPress={() => {
+                realm.write(() => {
+                  realm.delete(item)
+                })
+              }} ><Text>{"🗑️"}</Text></Pressable>
+          </View>
+        );
+      }} ></FlatList>
+    </SafeAreaView >
+  );
+}
 ```
 
-Wrap the component needing access to Realm (possibly your entire application) with the `RealmProvider` componenet.
-The `RealmProvider` also accepts Realm configuration properties.
+For a full fledged example, check out [our templates](https://github.com/realm/realm-js#template-apps-using-expo-for-react-native).
+
+## Realm Hooks
+### useQuery
+
+Returns a [Realm.Result] from a given type. This Hook will update on any changes to any object in the collection and return an empty array if the collection is empty.
+The result of this can be consumed directly by the `data` argument of any React Native VirtualizedList or FlatList.  If the component used for the list's `renderItem` prop is wrapped with [React.Memo], then only the modified object will re-render.
 
 ```tsx
-import RealmContext from './realm';
+const Component = () => {
+  // ObjectClass is a class extending Realm.Object should have been provided in the Realm Config.
+  // It is also possible to use the models name as a string( ex. "Object" ) if you are not using class based models.
+  const collection = useQuery(ObjectClass);
+
+  // The methods `sorted` and `filtered` should be wrapped in a useMemo.
+  const sortedCollection = useMemo(collection.sorted(), [collection]);
+
+  return (
+    <FlatList data={sortedCollection} renderItem={({ item }) => <Object item={item}/>
+  )
+}
+```
 
-const {RealmProvider} = RealmContext
+### useObject
+ Returns a [Realm.Object] for a given type and primary key.  The Hook will update on any changes to the properties on the returned object and return null if it either doesn't exist or has been deleted.
+```tsx
+const Component = ({someId}) => {
+  // ObjectClass is a class extending Realm.Object should have been provided in the Realm Config.
+  // It is also possible to use the models name as a string( ex. "Object" ) if you are not using class based models.
+  const object = useObject(ObjectClass, someId);
 
-function AppWrapper() {
+  return (
+    <View>
+      <Text>{object.name}</Text>
+    </View>
+  )
+}
+```
+## Setting Things Up
+### createRealmContext
+
+To get started with `@realm/react`, one must create a context object with `createRealmContext`.  The context object will contain a Realm Context Provider, which will have an open Realm as its context, and a set of hooks that access the Realm context.
```

Comment:
```suggestion
To get started with `@realm/react`, one must create a Context object with `createRealmContext`.  The Context object will contain a Realm Context Provider, which will have an open Realm as its context, and a set of hooks that access the Realm Context.
```

Reply:
I'm not quite sure what the different "context"s refer to here (I assume one is a general JS context, and one is a Realm one?).  I suggest we capitalize the names of any types that aren't native to JS.